gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/services/ApiServiceCockpitImpl.java

```diff
@@ -155,7 +155,12 @@ private ApiEntityResult updateApiEntity(String apiId, ImportSwaggerDescriptorEnt
         final SwaggerApiEntity api = swaggerService.createAPI(swaggerDescriptor, DefinitionVersion.V2);
         api.setPaths(null);
 
-        return ApiEntityResult.success(this.apiService.updateFromSwagger(apiId, api, swaggerDescriptor));
+        final Optional<String> result = checkContextPath(api);
+        if (result.isEmpty()) {
+            return ApiEntityResult.success(this.apiService.updateFromSwagger(apiId, api, swaggerDescriptor));
+        }
+
+        return ApiEntityResult.failure(result.get());
```

Comment:
⬇️ should be the same using Optional API

```suggestion
        return checkContextPath(api)
            .map(ApiEntityResult::failure)
            .orElseGet(() -> ApiEntityResult.success(apiService.update(apiId, api)));
```